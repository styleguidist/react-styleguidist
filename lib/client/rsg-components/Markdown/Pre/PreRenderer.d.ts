import React from 'react';
import { JssInjectedProps } from 'rsg-components/Styled';
export interface PreProps {
    className?: string;
    children: React.ReactNode;
}
declare type PrePropsWithClasses = JssInjectedProps & PreProps;
export declare const PreRenderer: React.FunctionComponent<PrePropsWithClasses>;
declare const _default: React.ComponentType<Pick<PrePropsWithClasses, "children" | "className">>;
export default _default;
