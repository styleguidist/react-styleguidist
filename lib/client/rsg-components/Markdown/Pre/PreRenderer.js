import "core-js/modules/es.array.index-of";
import "core-js/modules/es.object.assign";
import "core-js/modules/es.object.to-string";
import "core-js/modules/es.regexp.to-string";
import "core-js/modules/es.string.small";
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'clsx';
import Styled from 'rsg-components/Styled';
import prismTheme from '../../../styles/prismTheme';

var styles = function styles(_ref) {
  var space = _ref.space,
      color = _ref.color,
      fontSize = _ref.fontSize,
      fontFamily = _ref.fontFamily,
      borderRadius = _ref.borderRadius;
  return {
    pre: Object.assign({
      fontFamily: fontFamily.monospace,
      fontSize: fontSize.small,
      lineHeight: 1.5,
      color: color.base,
      whiteSpace: 'pre-wrap',
      wordWrap: 'normal',
      tabSize: 2,
      hyphens: 'none',
      backgroundColor: color.codeBackground,
      padding: [[space[1], space[2]]],
      border: [[1, color.codeBackground, 'solid']],
      borderRadius: borderRadius,
      marginTop: 0,
      marginBottom: space[2],
      overflow: 'auto'
    }, prismTheme({
      color: color
    }))
  };
};

export var PreRenderer = function PreRenderer(_ref2) {
  var classes = _ref2.classes,
      className = _ref2.className,
      children = _ref2.children;
  var classNames = cx(className, classes.pre);
  var isHighlighted = className && className.indexOf('lang-') !== -1;

  if (isHighlighted && children) {
    return /*#__PURE__*/React.createElement("pre", {
      className: classNames,
      dangerouslySetInnerHTML: {
        __html: children.toString()
      }
    });
  }

  return /*#__PURE__*/React.createElement("pre", {
    className: classNames
  }, children);
};
PreRenderer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};
export default Styled(styles)(PreRenderer);