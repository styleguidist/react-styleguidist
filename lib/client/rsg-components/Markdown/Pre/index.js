export { default } from 'rsg-components/Markdown/Pre/PreRenderer';
export * from 'rsg-components/Markdown/Pre/PreRenderer';