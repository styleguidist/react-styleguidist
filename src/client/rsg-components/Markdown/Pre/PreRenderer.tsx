import React from 'react';
import PropTypes from 'prop-types';
import cx from 'clsx';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';
import prismTheme from '../../../styles/prismTheme';
import * as Rsg from '../../../../typings';

const styles = ({ space, color, fontSize, fontFamily, borderRadius }: Rsg.Theme) => ({
	pre: {
		fontFamily: fontFamily.monospace,
		fontSize: fontSize.small,
		lineHeight: 1.5,
		color: color.base,
		whiteSpace: 'pre-wrap',
		wordWrap: 'normal',
		tabSize: 2,
		hyphens: 'none',
		backgroundColor: color.codeBackground,
		padding: [[space[1], space[2]]],
		border: [[1, color.codeBackground, 'solid']],
		borderRadius,
		marginTop: 0,
		marginBottom: space[2],
		overflow: 'auto',
		...prismTheme({ color }),
	},
});

export interface PreProps {
	className?: string;
	children: React.ReactNode;
}

type PrePropsWithClasses = JssInjectedProps & PreProps;

export const PreRenderer: React.FunctionComponent<PrePropsWithClasses> = ({
	classes,
	className,
	children,
}) => {
	const classNames = cx(className, classes.pre);

	const isHighlighted = className && className.indexOf('lang-') !== -1;
	if (isHighlighted && children) {
		return <pre className={classNames} dangerouslySetInnerHTML={{ __html: children.toString() }} />;
	}
	return <pre className={classNames}>{children}</pre>;
};

PreRenderer.propTypes = {
	classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
	className: PropTypes.string,
	children: PropTypes.any.isRequired,
};

export default Styled<PrePropsWithClasses>(styles)(PreRenderer);
