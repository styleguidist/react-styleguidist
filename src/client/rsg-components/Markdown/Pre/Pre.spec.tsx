import React from 'react';
import renderer from 'react-test-renderer';

import Pre from './index';

describe('Markdown Pre', () => {
	it('should render a pre', () => {
		const actual = renderer.create(<Pre>This is pre-formatted text.</Pre>);

		expect(actual.toJSON()).toMatchSnapshot();
	});

	it('should render highlighted code', () => {
		const code = '<button>OK</button>';
		const actual = renderer.create(<Pre className="lang-html">{code}</Pre>);

		expect(actual.toJSON()).toMatchSnapshot();
	});
});
