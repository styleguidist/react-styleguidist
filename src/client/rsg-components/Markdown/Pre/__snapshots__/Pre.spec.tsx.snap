// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Pre should render a pre 1`] = `
<pre
  className="rsg--pre-0"
>
  This is pre-formatted text.
</pre>
`;

exports[`Markdown Pre should render highlighted code 1`] = `
<pre
  className="lang-html rsg--pre-0"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<button>OK</button>",
    }
  }
/>
`;
